generators/typescript/asIs/tests/mock-server/withJson.ts

```diff
@@ -0,0 +1,146 @@
+import { HttpResponseResolver, passthrough } from "msw";
+
+import { fromJson, toJson } from "../../src/core/json";
+
+/**
+ * Creates a request matcher that validates if the request JSON body exactly matches the expected object
+ * @param expectedBody - The exact body object to match against
+ * @param resolver - Response resolver to execute if body matches
+ */
+export function withJson(expectedBody: unknown, resolver: HttpResponseResolver): HttpResponseResolver {
+    return async (args) => {
+        const { request } = args;
+
+        let clonedRequest: Request;
+        let actualBody: unknown;
+        try {
+            clonedRequest = request.clone();
+            actualBody = fromJson(await clonedRequest.text());
+        } catch (error) {
+            console.error("Error processing request body:", error);
+            return passthrough();
+        }
+
+        const mismatches = findMismatches(actualBody, expectedBody);
```

Comment:
The `findMatches` implementation is entirely custom here, right? 

I wonder if we should instead largely reference `lodash`'s `isEqual` function and include a comment that the implementation is directly inspired by `lodash`. That way, we're far less likely to have any weird edge cases here and can more easily keep the code in-sync if the source of truth is ever updated.

Thoughts?

Reply:
Understood - as long as we're confident this is _correct_. I imagine this could be a code smell to future SDK readers, but these are implementation details that we can change later on (if ever).